src/main/kotlin/org/elm/ide/actions/ElmExternalFormatAction.kt

```diff
@@ -49,14 +47,9 @@ class ElmExternalFormatAction : AnAction() {
             return
         }
 
-        FileDocumentManager.getInstance().saveAllDocuments()
         try {
-            ProgressManager.getInstance().runProcessWithProgressSynchronously<ProcessOutput, ExecutionException>({
-                elmFormat.reformatFile(project, ctx.elmVersion, virtualFile)
-            }, "Running elm-format on current file...", true, project)
-            // We want to refresh file synchronously only in unit test
-            // to get new text right after `reformatFile` call
-            VfsUtil.markDirtyAndRefresh(!isUnitTestMode, true, true, virtualFile)
+            val document = FileDocumentManager.getInstance().getDocument(virtualFile)!!
```

Comment:
The point of the `Context` object is to gather together everything that the action needs to successfully run so that you don't have to `!!` things. So I would either add the document to the context or do a check and early return at the beginning of `actionPerformed`.

Reply:
Good idea! I adjusted the `Context` object and pass it to the formatting method. Will use it also in my SaveHook impl.

Didn't look at it properly beforehand and thought it was something IDEA-internal.